int16 offset_x
int16 offset_y